### PR TITLE
PartitionReceiver: Implement invoke receiveHandler on timeout

### PR DESCRIPTION
**Pull request by SreeramGarlapati**
This will impact the EPH behavior - as we changed the default behavior for PartitionReceiver.
Fix for #191

### CODE REVIEW

**Comment by SreeramGarlapati** [comment]
@hmlam is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@sbcr is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@sjkwak is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@jbird is added to the review. #Closed

**Comment by SreeramGarlapati** [comment]
@JamesBirdsall is added to the review. #Closed

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java [hunk 1]
```diff
@@ -312,6 +328,10 @@ public void setReceiveHandler(final PartitionReceiveHandler receiveHandler)
 			}
 			else
 			{
+				if (this.isOnReceivePumpRunning)
+					throw new IllegalArgumentException(
+							"Unexpected value for parameter 'receiveHandler'. Close the existing running pump and then set the new handler");
```
Comment (hmlam) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java]: >Unexpected value for parameter 'receiveHandler' [](start = 8, length = 47)

we should just say only one receiveHandler can be registered with one receiver (pump running means we already set it once).

File: java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java [hunk 2]
```diff
@@ -291,17 +292,32 @@ public final long getEpoch()
...
 
 	/**
 	 * Register a receive handler that will be called when an event is available. A 
 	 * {@link PartitionReceiveHandler} is a handler that allows user to specify a callback
 	 * for event processing and error handling in a receive pump model. 
 	 * @param receiveHandler An implementation of {@link PartitionReceiveHandler}
+	 * @param invokeOnTimeout flag to indicate whether the {@link PartitionReceiveHandler#onReceive(Iterable)} should be invoked when the receive call times out
 	 */
-	public void setReceiveHandler(final PartitionReceiveHandler receiveHandler)
+	public void setReceiveHandler(final PartitionReceiveHandler receiveHandler, final boolean invokeOnTimeout)
```
Comment (hmlam) [inline java/azure-eventhubs/src/main/java/com/microsoft/azure/eventhubs/PartitionReceiver.java]: >invokeOnTimeout [](start = 91, length = 15)

wording of the argument is implying that we can get TimeoutException which is not really correct. I would suggest we say something more like "invokeOnNullEvents" to be more explicit.

**Comment by hmlam** [comment]
:clock1:

**Comment by JamesBirdsall** [comment]
Looks fine to me.

**Pull request merged**